subsys/bluetooth/host/gatt.c

```diff
@@ -3305,10 +3515,81 @@ static int gatt_read_multiple(struct bt_conn *conn,
 		net_buf_add_le16(buf, params->handles[i]);
 	}
 
-	return gatt_send(conn, buf, gatt_read_multiple_rsp, params, NULL);
+	return gatt_send(conn, buf, gatt_read_mult_rsp, params, NULL);
 }
+
+#if defined(CONFIG_BT_EATT)
+static void gatt_read_mult_vl_rsp(struct bt_conn *conn, u8_t err,
+				  const void *pdu, u16_t length,
+				  void *user_data)
+{
+	struct bt_gatt_read_params *params = user_data;
+	const struct bt_att_read_mult_vl_rsp *rsp;
+	struct net_buf_simple buf;
+
+	BT_DBG("err 0x%02x", err);
+
+	if (err || !length) {
+		if (err == BT_ATT_ERR_NOT_SUPPORTED) {
+			gatt_read_mult(conn, params);
+		}
+		params->func(conn, err, params, NULL, 0);
+		return;
+	}
+
+	net_buf_simple_init_with_data(&buf, (void *)pdu, length);
```

Comment:
Maybe add a code comment above this since otherwise it looks quite dangerous (casting `const void *` to `void *`). E.g. mention that it's safe to do since none of the `net_buf_simple` operations you do on the buffers modifies the data in it.

Reply:
Will do.